### PR TITLE
Feat/pause contract

### PR DESCRIPTION
**Pull request by oscgu**

### CODE REVIEW

File: src/PRS.sol [hunk 1]
```diff
@@ -100,6 +101,21 @@ contract PRS is Ownable, TaxableGame {
...
+        _pause();
+    }
+
+    // @notice Unpause contract
+    function unpauseContract() public onlyOwner {
+        _unpause();
+    }
+
+    // @notice Gets the current contract state
+    function contractState() public view onlyOwner returns (bool) {
```
Comment (faces-of-eth) [inline src/PRS.sol:115]: I believe you don't need this and can access paused() directly without consequence. Would remove this.

File: src/PRS.sol [hunk 2]
```diff
@@ -100,6 +101,21 @@ contract PRS is Ownable, TaxableGame {
         return game.entryFee;
     }
 
+    // @notice Pause contract incase of suspicious activity
+    function pauseContract() public onlyOwner {
```
Comment (faces-of-eth) [inline src/PRS.sol:105]: Maybe `pauseGame` or just `pause` ?

File: src/PRS.sol [hunk 3]
```diff
@@ -100,6 +101,21 @@ contract PRS is Ownable, TaxableGame {
...
+        _pause();
+    }
+
+    // @notice Unpause contract
+    function unpauseContract() public onlyOwner {
```
Comment (faces-of-eth) [inline src/PRS.sol:110]: Same comment as above

**Review by faces-of-eth — CHANGES_REQUESTED** [review]
Couple of suggestions for tweaking this. I'd also think about suspending all withdraws when contract is paused in TaxableGame

File: src/PRS.sol (hunk 1 above)
Comment (oscgu) [reply src/PRS.sol]: [cecbb6d](https://github.com/dopedao/prs/pull/3/commits/cecbb6ddf299bf18e9c278a462da8f9ca5c47c05)

File: src/PRS.sol (hunk 2 above)
Comment (oscgu) [reply src/PRS.sol]: [78016d3](https://github.com/dopedao/prs/pull/3/commits/78016d3228b27af4c362a3181bb383b944930daf)
[f617318](https://github.com/dopedao/prs/pull/3/commits/f617318b2cd4ace827bbce01b55c7d3da8330b3e)

File: src/PRS.sol (hunk 3 above)
Comment (oscgu) [reply src/PRS.sol]: [78016d3](https://github.com/dopedao/prs/pull/3/commits/78016d3228b27af4c362a3181bb383b944930daf)
[f617318](https://github.com/dopedao/prs/pull/3/commits/f617318b2cd4ace827bbce01b55c7d3da8330b3e)

**Review by faces-of-eth — APPROVED** [review]
lgtm

**Pull request merged**